src/jarque_bera.jl

```diff
@@ -0,0 +1,69 @@
+# jarque_bera.jl
+# Jarque-Bera goodness-of-fit test
+#
+# Copyright (C) 2017   Benjamin Born
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export JarqueBeraTest
+
+immutable JarqueBeraTest <: HypothesisTest
+    n       ::Int           # number of observations
+    JB      ::Float64       # test statistic
+    m3      ::Float64       # skewness
+    m4      ::Float64       # excess kurtosis
+end
+
+"""
+    JarqueBeraTest(y)
+
+Compute the Jarque-Bera statistic to test the null hypothesis that a series `y` comes from a normal distribution.
+
+Note that the p-values might be distorted in small samples.
+
+External links
+
+* [Jarque-Bera test on Wikipedia](https://en.wikipedia.org/wiki/Jarque–Bera_test)
+"""
+
+function JarqueBeraTest{T<:Real}(y::AbstractVector{T})
+    n = length(y)
+    m1 = sum(y)/n
+    m2 = sum((y - m1).^2)/n
+    m3 = sum((y - m1).^3)/n
+    m4 = sum((y - m1).^4)/n
```

Comment:
You can simplify this to make a single pass over the input:
```julia
n = length(y)
S = Base.promote_op(/, T, typeof(n))
m1 = m2 = m3 = m4 = zero(S)
@inbounds for yi in y
    m1 += yi / n
    m2 += yi^2 / n
    m3 += yi^3 / n
    m4 += yi^4 / n
end
```
This is 4n divisions, which is not great, but it's better than possibly overflowing if we wait until the end to divide. It should still be better than multiple passes over the input, at any rate.